[BUG] Automated tests fail locally

### Problem description

I've cloned the current main branch (commit af30a253fb1ec45fd236e46d46230f81f0f21c8e) and run `hatch test`. There are two test failures, seemingly due to actual bugs in the code (yet CI passes and seems to be running the same suite?).

### Code sample

```text
================================================ test session starts =================================================
platform linux -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0

Running tests with petrofit version 0.5.2.dev66+gaf30a25.
Running tests in tests.

Date: 2024-11-27T21:03:31

Platform: Linux-6.11.8-200.fc40.x86_64-x86_64-with-glibc2.39

Executable: /home/txwx36/.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/bin/python3

Full Python Version: 
3.12.7 (main, Oct  1 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 2.1.3
Scipy: 1.14.1
Matplotlib: 3.9.2
h5py: not available
scikit-image: 0.24.0


rootdir: /home/txwx36/code/petrofit
configfile: pyproject.toml
plugins: anyio-4.6.2.post1, astropy-header-0.2.2, mock-3.14.0, rerunfailures-14.0, xdist-3.6.1
collected 28 items                                                                                                   

tests/test_fitting.py F..F                                                                                     [ 14%]
tests/test_petrosian.py ....                                                                                   [ 28%]
tests/test_photometry.py ....                                                                                  [ 42%]
tests/test_segmentation.py ........                                                                            [ 71%]
tests/test_utils.py ........                                                                                   [100%]

====================================================== FAILURES ======================================================
___________________________________________ test_psf_convolved_image_model ___________________________________________

    def test_psf_convolved_image_model():
        """Test fitting and PSF convolution"""
    
        # Make model:
        imsize = 300
    
        sersic_model = models.Sersic2D(
            amplitude=1,
            r_eff=25,
            n=2,
            x_0=imsize / 2,
            y_0=imsize / 2,
            ellip=0,
            theta=0,
            bounds={
                "amplitude": (0.0, None),
                "r_eff": (0, None),
                "n": (0, 10),
                "ellip": (0, 1),
                "theta": (-2 * np.pi, 2 * np.pi),
            },
        )
    
        # Make model image
        image = pf.model_to_image(sersic_model, imsize)
    
        # Make a PSF
        x_grid, y_grid = pf.make_grid(51, factor=1)
        PSF = models.Moffat2D(x_0=25.0, y_0=25.0)(x_grid, y_grid)
        PSF /= PSF.sum()
    
        # Make a PSF image using model image and PSF
        psf_sersic_image = convolve(image, PSF)
    
        # Make a PSFConvolvedModel2D
        psf_sersic_model = pf.PSFConvolvedModel2D(sersic_model, psf=PSF, oversample=None)
        psf_sersic_model.fixed["psf_pa"] = True
    
        # Make a PSFConvolvedModel2D image
        psf_sersic_model_image = pf.model_to_image(psf_sersic_model, imsize)
    
        # Compare the PSF image to PSFConvolvedModel2D image
        error_arr = abs(psf_sersic_model_image - psf_sersic_image) / psf_sersic_image
        assert np.max(error_arr) < 0.01  # max error less than 1% error
    
        # Test fitting
    
        # Change params with offset
        psf_sersic_model.r_eff = 23
        psf_sersic_model.x_0 = 0.6 + imsize / 2
        psf_sersic_model.y_0 = 0.1 + imsize / 2
        psf_sersic_model.n = 3
    
        # Fit
>       fitted_model, fit_info = pf.fit_model(
            psf_sersic_image,
            psf_sersic_model,
            maxiter=10000,
            epsilon=1.4901161193847656e-10,
            acc=1e-9,
        )

tests/test_fitting.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
modeling/fitting.py:112: in fit_model
    fitted_model = fitter_instance(
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:303: in wrapper
    return func(self, model, x, y, z=z, **kwargs)
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:1434: in __call__
    init_values, fitparams, cov_x = self._run_fitter(
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:1608: in _run_fitter
    self.fit_info = optimize.least_squares(
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/scipy/optimize/_lsq/least_squares.py:832: in least_squares
    f0 = fun_wrapped(x0)
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/scipy/optimize/_lsq/least_squares.py:830: in fun_wrapped
    return np.atleast_1d(fun(x, *args, **kwargs))
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:1184: in objective_function
    value = np.ravel(model.evaluate(*inputs, *fps) - meas)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PSFConvolvedModel2D(amplitude=1., r_eff=23., n=3., x_0=150.6, y_0=150.1, ellip=0., theta=0., psf_pa=0.)>
x = array([  0.,   1.,   2., ..., 297., 298., 299.]), y = array([  0.,   0.,   0., ..., 299., 299., 299.])
params = (np.float64(1.0), np.float64(23.0), np.float64(3.0), np.float64(150.6), np.float64(150.1), np.float64(1e-10), ...)
kwargs = {}
sub_model_params = [np.float64(1.0), np.float64(23.0), np.float64(3.0), np.float64(150.6), np.float64(150.1), np.float64(1e-10), ...]
psf_p = np.float64(0.0), i = array([  0,   1,   2, ..., 297, 298, 299])
j = array([  0,   0,   0, ..., 299, 299, 299]), is_subgrid_oversample = False, psf_factor = 1, grid_factor = 1
grid_to_psf_factor = 1, grid_size = np.int64(300)
grid_range = (np.float64(0.0), np.float64(299.0), np.float64(0.0), np.float64(299.0))

    def evaluate(self, x, y, *params, **kwargs):
    
        # Extract sub-model params as well as `psf_p`
        *sub_model_params, psf_p = params
    
        # Prepare image indices
        i = (x - x.min()).astype(int)
        j = (y - y.min()).astype(int)
    
        # Is oversampling sub-grid based
        is_subgrid_oversample = isinstance(self.oversample, tuple)
    
        # Compute image size and oversampling factor
        # ------------------------------------------
        psf_factor = (
            self._get_psf_factor()
        )  # Oversampling factor of PSF compared to data
    
        # Grid oversample factor compared to data:
        # If the oversampling is subgrid, then the main grid is at PSF oversample.
        # If the oversampling is an int, the main grid is at oversample factor.
        grid_factor = (
            psf_factor if is_subgrid_oversample else self._get_oversample_factor()
        )
    
        grid_to_psf_factor = (
            grid_factor // psf_factor
        )  # Oversampling factor of grid compared to PSF
    
        # Grid size params:
        grid_size = max([i.max(), j.max()]) + 1  # size = max_index + 1
        grid_range = (x.min(), x.max(), y.min(), y.max())  # Main grid bounds
    
        # Make main grid
        # --------------
        # Make the main sampling grid
        if (
            grid_size == self._cached_grid_size
            and grid_factor == self._cached_grid_factor
            and grid_range == self._cache_grid_range
        ):
            # If the sampling gird cached
            main_grid = self._cached_grid
        else:
            # Else make a sampling gird
            main_grid = make_grid(
                grid_size, origin=(x.min(), y.min()), factor=grid_factor
            )
            # Cache Grid
            if self.cache_grid:
                self._cached_grid = main_grid
                self._cached_grid_size = grid_size
                self._cached_grid_factor = grid_factor
                self._cache_grid_range = grid_range
    
        # Split main grid to x and y components
        x_grid, y_grid = main_grid
    
        # Main Model Image
        # ----------------
        # Construct main model image by sampling sub-model
        model_image = self._model.evaluate(x_grid, y_grid, *sub_model_params)
    
        # Oversampling
        # ------------
        if not is_subgrid_oversample and grid_to_psf_factor > 1:
            # If the oversample factor is an int, block reduce the image to PSF resolution
            model_image = (
                block_reduce(model_image, grid_to_psf_factor) / grid_to_psf_factor**2
            )
    
        elif is_subgrid_oversample:
            # If the oversample is a window, compute pixel values for that window
            # and block reduce the image to PSF resolution.
    
            # Load the window params
            sub_grid_x0, sub_grid_y0, sub_grid_size, sub_grid_factor = self.oversample
            sub_grid_to_psf_factor = sub_grid_factor // psf_factor
    
            if sub_grid_to_psf_factor > 1:
                # If the center of the window is a parameter name, extract its value
                if isinstance(sub_grid_x0, str):
                    assert (
                        sub_grid_x0 in self._model.param_names
                    ), "oversample param '{}' is not in the wrapped model param list".format(
                        sub_grid_x0
                    )
                    idx = self._model.param_names.index(sub_grid_x0)
                    sub_grid_x0 = sub_model_params[idx][0]
    
                if isinstance(sub_grid_y0, str):
                    assert (
                        sub_grid_y0 in self._model.param_names
                    ), "oversample param '{}' is not in the wrapped model param list".format(
                        sub_grid_y0
                    )
                    idx = self._model.param_names.index(sub_grid_y0)
                    sub_grid_y0 = sub_model_params[idx][0]
    
                # Compute the corner of the sub-grid
                sub_grid_origin = (
                    np.round(sub_grid_x0) - sub_grid_size // 2,
                    np.round(sub_grid_y0) - sub_grid_size // 2,
                )
    
                # Make an oversampled sub-grid for window
                x_sub_grid, y_sub_grid = make_grid(
                    sub_grid_size, origin=sub_grid_origin, factor=sub_grid_factor
                )
    
                # Sample the sub-model onto the sub-grid
                sub_model_oversampled_image = self._model.evaluate(
                    x_sub_grid, y_sub_grid, *sub_model_params
                )
    
                # Block reduce the window to the psf resolution
                sub_model_image = (
                    block_reduce(sub_model_oversampled_image, sub_grid_to_psf_factor)
                    / sub_grid_to_psf_factor**2
                )
    
                # Compute window indices in main image frame at data resolution first
                i_sub_min = int(np.round(sub_grid_origin[0]))
                j_sub_min = int(np.round(sub_grid_origin[1]))
                i_sub_max = i_sub_min + sub_grid_size
                j_sub_max = j_sub_min + sub_grid_size
    
                # Clip window indices
                if i_sub_min < 0:
                    i_sub_min = 0
                if j_sub_min < 0:
                    j_sub_min = 0
                if i_sub_max > i.max():
                    i_sub_max = i.max() + 1
                if j_sub_max > j.max():
                    j_sub_max = j.max() + 1
    
                # Convert to PSF resolution
                i_sub_min *= psf_factor
                j_sub_min *= psf_factor
                i_sub_max *= psf_factor
                j_sub_max *= psf_factor
    
                # Add oversampled window to image
                model_image[j_sub_min:j_sub_max, i_sub_min:i_sub_max] = sub_model_image
    
        # PSF convolve
        # ------------
        if self.psf is not None:
            psf = self.psf
>           if psf_p[0] != 0:
E           IndexError: invalid index to scalar variable.

modeling/models.py:503: IndexError
_________________________________________________ test_psf_sampling __________________________________________________

galfit_psf_images = {0: array([[0.00056343, 0.0005715 , 0.00057968, ..., 0.0005715 , 0.00056343,
        0.00055546],
       [0.0005715 , ...00055546],
       [0.00055546, 0.00056335, 0.00057133, ..., 0.00056335, 0.00055546,
        0.00054768]], dtype='>f4')}
psf_image = array([[2.1726346e-06, 3.4357931e-06, 4.8513293e-06, ..., 2.5688851e-06,
        3.0904168e-06, 3.1844179e-06],
      ... [1.5745794e-06, 1.7172894e-06, 1.9483414e-06, ..., 1.1671426e-05,
        1.4433384e-05, 1.3759987e-05]], dtype='>f4')

    def test_psf_sampling(galfit_psf_images, psf_image):
        for psf_index, image in galfit_psf_images.items():
            base_model = models.Sersic2D(
                amplitude=0.020625826413226116,
                r_eff=30,
                n=4,
                x_0=99.0,
                y_0=99.0,
                ellip=0,
                theta=0.0,
                bounds=pf.get_default_sersic_bounds(),
                fixed={
                    "x_0": False,
                    "y_0": False,
                    "n": True,
                    "r_eff": True,
                    "ellip": True,
                    "theta": True,
                },
            )
    
            PSF = psf_image if psf_index > 0 else None
            psf_oversample = psf_index if psf_index > 0 else None
            model = pf.PSFConvolvedModel2D(
                base_model,
                psf=PSF,
                oversample=(
                    "x_0",
                    "y_0",
                    100,
                    psf_oversample * 5 if PSF is not None else 50,
                ),
                psf_oversample=psf_oversample if PSF is not None else None,
            )
            model.fixed.update({"psf_pa": True})
    
>           fitted_model, fit = pf.fit_model(
                image,
                model,
                acc=1e-12,
            )

tests/test_fitting.py:247: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
modeling/fitting.py:112: in fit_model
    fitted_model = fitter_instance(
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:303: in wrapper
    return func(self, model, x, y, z=z, **kwargs)
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:1434: in __call__
    init_values, fitparams, cov_x = self._run_fitter(
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:1608: in _run_fitter
    self.fit_info = optimize.least_squares(
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/scipy/optimize/_lsq/least_squares.py:832: in least_squares
    f0 = fun_wrapped(x0)
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/scipy/optimize/_lsq/least_squares.py:830: in fun_wrapped
    return np.atleast_1d(fun(x, *args, **kwargs))
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/astropy/modeling/fitting.py:1184: in objective_function
    value = np.ravel(model.evaluate(*inputs, *fps) - meas)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <PSFConvolvedModel2D(amplitude=0.02062583, r_eff=30., n=4., x_0=99., y_0=99., ellip=0., theta=0., psf_pa=0.)>
x = array([  0.,   1.,   2., ..., 197., 198., 199.]), y = array([  0.,   0.,   0., ..., 199., 199., 199.])
params = (np.float64(0.020625826413226116), np.float64(30.0), np.float64(4.0), np.float64(99.0), np.float64(99.0), np.float64(0.0), ...)
kwargs = {}
sub_model_params = [np.float64(0.020625826413226116), np.float64(30.0), np.float64(4.0), np.float64(99.0), np.float64(99.0), np.float64(0.0), ...]
psf_p = np.float64(0.0), i = array([  0,   1,   2, ..., 197, 198, 199])
j = array([  0,   0,   0, ..., 199, 199, 199]), is_subgrid_oversample = True, psf_factor = 1, grid_factor = 1
grid_to_psf_factor = 1, grid_size = np.int64(200)
grid_range = (np.float64(0.0), np.float64(199.0), np.float64(0.0), np.float64(199.0))

    def evaluate(self, x, y, *params, **kwargs):
    
        # Extract sub-model params as well as `psf_p`
        *sub_model_params, psf_p = params
    
        # Prepare image indices
        i = (x - x.min()).astype(int)
        j = (y - y.min()).astype(int)
    
        # Is oversampling sub-grid based
        is_subgrid_oversample = isinstance(self.oversample, tuple)
    
        # Compute image size and oversampling factor
        # ------------------------------------------
        psf_factor = (
            self._get_psf_factor()
        )  # Oversampling factor of PSF compared to data
    
        # Grid oversample factor compared to data:
        # If the oversampling is subgrid, then the main grid is at PSF oversample.
        # If the oversampling is an int, the main grid is at oversample factor.
        grid_factor = (
            psf_factor if is_subgrid_oversample else self._get_oversample_factor()
        )
    
        grid_to_psf_factor = (
            grid_factor // psf_factor
        )  # Oversampling factor of grid compared to PSF
    
        # Grid size params:
        grid_size = max([i.max(), j.max()]) + 1  # size = max_index + 1
        grid_range = (x.min(), x.max(), y.min(), y.max())  # Main grid bounds
    
        # Make main grid
        # --------------
        # Make the main sampling grid
        if (
            grid_size == self._cached_grid_size
            and grid_factor == self._cached_grid_factor
            and grid_range == self._cache_grid_range
        ):
            # If the sampling gird cached
            main_grid = self._cached_grid
        else:
            # Else make a sampling gird
            main_grid = make_grid(
                grid_size, origin=(x.min(), y.min()), factor=grid_factor
            )
            # Cache Grid
            if self.cache_grid:
                self._cached_grid = main_grid
                self._cached_grid_size = grid_size
                self._cached_grid_factor = grid_factor
                self._cache_grid_range = grid_range
    
        # Split main grid to x and y components
        x_grid, y_grid = main_grid
    
        # Main Model Image
        # ----------------
        # Construct main model image by sampling sub-model
        model_image = self._model.evaluate(x_grid, y_grid, *sub_model_params)
    
        # Oversampling
        # ------------
        if not is_subgrid_oversample and grid_to_psf_factor > 1:
            # If the oversample factor is an int, block reduce the image to PSF resolution
            model_image = (
                block_reduce(model_image, grid_to_psf_factor) / grid_to_psf_factor**2
            )
    
        elif is_subgrid_oversample:
            # If the oversample is a window, compute pixel values for that window
            # and block reduce the image to PSF resolution.
    
            # Load the window params
            sub_grid_x0, sub_grid_y0, sub_grid_size, sub_grid_factor = self.oversample
            sub_grid_to_psf_factor = sub_grid_factor // psf_factor
    
            if sub_grid_to_psf_factor > 1:
                # If the center of the window is a parameter name, extract its value
                if isinstance(sub_grid_x0, str):
                    assert (
                        sub_grid_x0 in self._model.param_names
                    ), "oversample param '{}' is not in the wrapped model param list".format(
                        sub_grid_x0
                    )
                    idx = self._model.param_names.index(sub_grid_x0)
>                   sub_grid_x0 = sub_model_params[idx][0]
E                   IndexError: invalid index to scalar variable.

modeling/models.py:441: IndexError
================================================== warnings summary ==================================================
../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/_pytest/config/__init__.py:1441
  /home/txwx36/.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: doctest_plus
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../../.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/_pytest/config/__init__.py:1441
  /home/txwx36/.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/_pytest/config/__init__.py:1441: PytestConfigWarning: Unknown config option: text_file_format
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

petrofit/tests/test_petrosian.py::test_petrosian
  /home/txwx36/code/petrofit/petrofit/tests/test_petrosian.py:107: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_petrosian.py::test_petrosian
  /home/txwx36/code/petrofit/petrofit/tests/test_petrosian.py:110: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_petrosian.py::test_petrosian
  /home/txwx36/code/petrofit/petrofit/tests/test_petrosian.py:113: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_petrosian.py::test_corrections
  /home/txwx36/code/petrofit/petrofit/tests/test_petrosian.py:148: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_petrosian.py: 791 warnings
petrofit/tests/test_photometry.py: 15 warnings
petrofit/tests/test_segmentation.py: 120 warnings
  /home/txwx36/code/petrofit/petrofit/photometry.py:211: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    aperture_area = float(np.round(aperture_area, 6))

petrofit/tests/test_petrosian.py: 791 warnings
petrofit/tests/test_photometry.py: 15 warnings
petrofit/tests/test_segmentation.py: 120 warnings
  /home/txwx36/code/petrofit/petrofit/photometry.py:212: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
    photometric_value = float(np.round(photometric_value, 6))

petrofit/tests/test_photometry.py::test_plot_apertures
  /home/txwx36/code/petrofit/petrofit/tests/test_photometry.py:18: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_photometry.py::test_plot_apertures
  /home/txwx36/code/petrofit/petrofit/tests/test_photometry.py:22: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_photometry.py::test_plot_apertures
  /home/txwx36/code/petrofit/petrofit/tests/test_photometry.py:26: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_photometry.py::test_plot_apertures
  /home/txwx36/code/petrofit/petrofit/tests/test_photometry.py:30: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_photometry.py::test_radial_photometry
  /home/txwx36/code/petrofit/petrofit/tests/test_photometry.py:67: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_segmentation.py::test_make_catalog_no_deblend
petrofit/tests/test_segmentation.py::test_plot_segments
petrofit/tests/test_segmentation.py::test_plot_segments
  /home/txwx36/.local/share/hatch/env/virtual/petrofit/q4Ud3_eR/hatch-test.py3.12/lib64/python3.12/site-packages/matplotlib/cbook.py:733: DeprecationWarning: __array__ implementation doesn't accept a copy keyword, so passing copy=False failed. __array__ must implement 'dtype' and 'copy' keyword arguments.
    x = np.array(x, subok=True, copy=copy)

petrofit/tests/test_segmentation.py::test_make_catalog_no_deblend
  /home/txwx36/code/petrofit/petrofit/tests/test_segmentation.py:82: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_segmentation.py::test_plot_segments
  /home/txwx36/code/petrofit/petrofit/tests/test_segmentation.py:171: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_segmentation.py::test_plot_segments
  /home/txwx36/code/petrofit/petrofit/tests/test_segmentation.py:174: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_segmentation.py::test_plot_segments
  /home/txwx36/code/petrofit/petrofit/tests/test_segmentation.py:177: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

petrofit/tests/test_segmentation.py::test_source_photometry
  /home/txwx36/code/petrofit/petrofit/tests/test_segmentation.py:211: UserWarning: FigureCanvasAgg is non-interactive, and thus cannot be shown
    plt.show()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================== short test summary info ===============================================
FAILED tests/test_fitting.py::test_psf_convolved_image_model - IndexError: invalid index to scalar variable.
FAILED tests/test_fitting.py::test_psf_sampling - IndexError: invalid index to scalar variable.
==================================== 2 failed, 26 passed, 1871 warnings in 18.01s ====================================

```


### Environment

<details>
### Package Information

| Package  |       Version        |
|:---------|---------------------:|
| petrofit | 0.5.2.dev66+gaf30a25 |

### Dependency Information

|        Package        | Version |
|:----------------------|--------:|
| astropy               |   7.0.0 |
| cython                |  3.0.11 |
| extension-helpers     |   1.2.0 |
| hatch                 |  1.13.0 |
| ipywidgets            |   8.1.5 |
| jupyter               |   1.1.1 |
| matplotlib            |   3.9.2 |
| notebook              |   7.2.2 |
| numpy                 |   2.1.3 |
| photutils             |   2.0.2 |
| pytest-astropy-header |   0.2.2 |
| pyyaml                |   6.0.2 |
| regions               |    0.10 |
| scikit-image          |  0.24.0 |
| scikit-learn          |   1.5.2 |
| scipy                 |  1.14.1 |

### Build Tools Information

|  Package   | Version |
|:-----------|--------:|
| hatch      |  1.13.0 |
| pip        |  23.3.2 |
| setuptools |  75.6.0 |

### Platform Information

|         |                               |
|:--------|------------------------------:|
| Linux   | 6.11.8-200.fc40.x86_64-x86_64 |
| CPython |                        3.12.7 |

</details>

### Anything else?

_No response_

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -25,6 +25,11 @@ These significant changes are coming in light of the pyOpenSci review process. T
 
 - Add Generative AI Policy. #231 
 
+**General bug fixes and small changes**
+
+- Fixes issues with PSFConvolvedModel2D that came up in light of the latest astropy version. #237
+
+
 ## Version 0.5.1
 *August 16th 2024*
 

--- a/petrofit/modeling/models.py
+++ b/petrofit/modeling/models.py
@@ -438,7 +438,7 @@ class PSFConvolvedModel2D(FittableModel):
                         sub_grid_x0
                     )
                     idx = self._model.param_names.index(sub_grid_x0)
-                    sub_grid_x0 = sub_model_params[idx][0]
+                    sub_grid_x0 = sub_model_params[idx][0] if isinstance(sub_model_params[idx], (list, np.ndarray)) else sub_model_params[idx]
 
                 if isinstance(sub_grid_y0, str):
                     assert (
@@ -447,7 +447,7 @@ class PSFConvolvedModel2D(FittableModel):
                         sub_grid_y0
                     )
                     idx = self._model.param_names.index(sub_grid_y0)
-                    sub_grid_y0 = sub_model_params[idx][0]
+                    sub_grid_y0 = sub_model_params[idx][0] if isinstance(sub_model_params[idx], (list, np.ndarray)) else sub_model_params[idx]
 
                 # Compute the corner of the sub-grid
                 sub_grid_origin = (
@@ -500,8 +500,10 @@ class PSFConvolvedModel2D(FittableModel):
         # ------------
         if self.psf is not None:
             psf = self.psf
-            if psf_p[0] != 0:
-                psf = rotate(psf, psf_p[0], reshape=False)
+            
+            psf_p = psf_p[0] if isinstance(psf_p, (list, np.ndarray)) else psf_p 
+            if psf_p != 0:
+                psf = rotate(psf, psf_p, reshape=False)
             model_image = convolve(model_image, psf, mode="same")
 
         if psf_factor > 1:

